cosa/analyzers/mcsolver.py

```diff
@@ -232,16 +237,22 @@ def _add_assertion(self, solver, formula, comment=None):
             for v in formula_fv:
                 if v in solver.smt2vars:
                     continue
-                
+
+                if any(char in "[]$" for char in v.symbol_name()):
```

Comment:
it is good practice to define a constant on the top of the file instead of use them locally. I suggest to define a string constant ESCAPE_CHARS_SMTLIB = "[]$".